### DISCUSSION TITLE
Federation ID pictogram

### DISCUSSION THREAD

**Discussion by dpc** [fedimint/fedimint#3469] [General]
It's hard to visually recognize a long string of hex values. When the UI of something shows me the federation ID, I have no idea which federation did it belong to (if I'm interacting with multiple ones).

It might be of help if we encode graphically for cursory visually inspection. It's not meant to replace full verification for trustworthiness, but only a hint.

Maybe as one of these randomized avatar/image schemes would do, or some kind of emojis.

Possibly we should display both full encoding and short emojis next to each other in most places.

Notably Matrix uses [emoji verification](https://matrix.org/docs/guides/implementing-more-advanced-e-2-ee-features-such-as-cross-signing#1-implement-emoji-verification-sas) for confirming public keys in e2e encryption. Maybe we could even address the reasonably secure verification of the Federation ID here as well.

![image](https://user-images.githubusercontent.com/9209/232850605-b8221474-7b5a-4513-ac89-344ad062c4c9.png)

https://wiki.physik.fu-berlin.de/it/services:matrix:encryption

**Comment by elsirion** [comment]
I'm wondering where this should be defined. Maybe we want a document describing Fedimint Client best practices (e.g. letting users compare federation IDs after joining) and standards for representation of certain data? Should this document live in this repo or a separate one? Thoughts @sectore @wbobeirne @skylerwww?
